client/blocks/dismissible-card/index.jsx

```diff
@@ -27,7 +29,13 @@ function DismissibleCard( { className, highlight, temporary, onClick, preference
 	return (
 		<Card className={ className } highlight={ highlight }>
 			<QueryPreferences />
-			<Gridicon icon="cross" className="dismissible-card__close-icon" onClick={ handleClick } />
+			<button
+				className="dismissible-card__close-icon"
+				onClick={ handleClick }
+				aria-label={ __( 'Dismiss' ) }
```

Comment:
I paired with @noahtallen on some of the other tests here, and Noah shared this option as one he'd seen in other recent commits. Is there an explicit reason that would make it clear when to use that library? Or are there any resources that could help us know which library to use in the future? 

In the meantime, I'll go ahead and change it to `useTranslate` from `i18n-calypso`. 🙂

Reply:
Good question, and I might as well have clarified previously of course 😅 

In general, `i18n-calypso` is used when doing any changes to the main entry point, which seems to be the case for this instance - this block is used only in the main entry point. In the stepper, or gutenboarding entry points, we use the other packages, since they are wrappers around `@wordpress/i18n` and in that sense are closer to the WordPress packages and infrastructure. 

Another point is that we have a bunch of additional functionality that's supported for i18n-calypso, but not for the `@wordpress/i18n`-based packages. A few examples that come to mind are the Community Translator and Empathy Mode. Also, I recall there was something off with how we index the translations, but I'd defer to the @Automattic/i18n folks.

Ideally, we should be using the same i18n package for everything, and that would likely be `@automattic/react-i18n`, but I'm not sure how far we are in the preparation for the migration. @yuliyan and the rest of @Automattic/i18n what's the current status? What would you folks recommend us using for localizing new strings in Calypso nowadays?

Reply:
The migration to `@wordpress/react-i18n` is still on our roadmap. However it's been a lower priority one and we haven't touched it in a while, due to lack of team capacity. Last time I've looked into the subject, there was a major blocker (which I believe is still present to this day) for migrating our i18n extensions (Community Translator and Empathy Mode) related to how translate functions are disconnected from the string utility functions (`sprintf`, `createInterpolateElement`) that are usually used with in the `@wordpress` eco system.

Currently, `i18n-calypso` and `@wordpress/react-i18n` work in sync on the main and stepper entry points, sharing the same translations data. We also support translatable string extraction from the source code for the translation calls from either of the libraries. So, in general, either should work just fine for rendering translations. However, our i18n extensions (e.g. Community Translator, Empathy Mode) don't have support for `@wordpress/react-i18n`.

As @tyxla outlined above, `i18n-calypso` is the translation library that's being used in the main entry point and packages that are only being used within Calypso, so I'd also recommend to keep using it until we're ready for the migration.